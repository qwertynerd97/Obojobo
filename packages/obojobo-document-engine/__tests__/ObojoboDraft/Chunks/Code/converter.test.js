import Converter from 'ObojoboDraft/Chunks/Code/converter'

jest.mock('src/scripts/oboeditor/util/text-util')

describe('Code Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			key: 'mockKey',
			type: 'mockType',
			data: {
				get: () => {
					return null
				}
			},
			nodes: [
				{
					text: 'mockCode',
					data: {
						get: () => {
							return {}
						}
					},
					nodes: [
						{
							leaves: [
								{
									text: 'mockCode',
									marks: [
										{
											type: 'b',
											data: {}
										}
									]
								}
							]
						}
					]
				}
			]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				textGroup: [
					{
						data: { indent: 1 },
						text: { value: 'mockCode' }
					},
					{
						text: { value: 'mockCode2' }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
