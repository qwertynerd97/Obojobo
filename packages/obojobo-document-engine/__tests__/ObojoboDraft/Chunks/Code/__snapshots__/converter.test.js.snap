// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "data": Object {
    "content": Object {},
  },
  "key": "mockKey",
  "nodes": Array [
    Object {
      "data": Object {
        "content": Object {
          "indent": 1,
        },
      },
      "nodes": Array [
        Object {
          "leaves": undefined,
          "object": "text",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.Code.CodeLine",
    },
    Object {
      "data": Object {
        "content": Object {
          "indent": 0,
        },
      },
      "nodes": Array [
        Object {
          "leaves": undefined,
          "object": "text",
        },
      ],
      "object": "block",
      "type": "ObojoboDraft.Chunks.Code.CodeLine",
    },
  ],
  "object": "block",
  "type": "mockType",
}
`;

exports[`Code Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "textGroup": Array [
      Object {
        "data": Object {
          "indent": undefined,
        },
        "text": Object {
          "styleList": Array [],
          "value": "mockCode",
        },
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
